docs/design/2018-10-24-view-support.md

```diff
@@ -0,0 +1,49 @@
+# Proposal: Implement View Feature
+
+- Author(s):     AndrewDi
+- Last updated:  2018-10-24
+- Discussion at: https://github.com/pingcap/tidb/issues/7974
+
+## Abstract
+This proposal proposes to implement view feature in TiDB, aimed to make sql more easy to write.
+
+## Background
+A database view is a searchable object in a database that is defined by a query.  Though a view doesn’t store data, some refer to a views as “virtual tables,” you can query a view like you can a table.  A view can combine data from two or more table, using joins, and also just contain a subset of information.  This makes them convenient to abstract, or hide, complicated queries.
+
+Below is a visual depiction of a view:
+  ![AnatomyOfAview](.imgs/view.png)
+  
+A view is created from a query using the CREATE OR REPLACE VIEW command.In the example below we are creating a PopularBooks view based of a query which selects all Books that have the IsPopular field checked.  Following is the query:
+`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+
+Once a view is created you can used then as you would any table in a SELECT statement.  For example, to list all the popular book titles ordered by author you could write:  
+`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+
+In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
+
+## Proposal
+I implement `create or replace` command to create view with select statement, currently I reuse `drop table` to drop a view,`show full tables` can also display view list with different table types.`show create table` command can also regenerated a view create command.
+
+## Rationale
+This proposal has no effect on the rationale.
+
+## Compatibility
+This proposal has no effect on the compatibility.
+
+## Implementation
+First we need to introduce `TableType` for `TableInfo`, currently I define two kind of table, which is `BaseTable` and `View`,by default if we use `CREATE TABLE` command to create a table ,TiDB create a BaseTable `TableInfo`,if we use `CREATE VIEW` command to a create a table,TiDB create a View `TableInfo`.
+
+In order to store the VIEW's select text,we also need to add attribute `ViewSelectStmt string json:"view_select_stmt"` store select statement.
+
+Following is the main implementation details:  
+1. ast/ddl.go  
+I do check both view name and select statement,viewname is a tablename object.
+2. planner/core/planbuilder.go  
+The code execute select and return `expression.columns` within `plan.Schema()`.
+3. ddl_api.go  
+We convert `expression.columns` into `table.Column` and reuse function `buildTableInfo` to build view
+4. logical_plan_builder.go  
+Everytime we select from a view,convert view to select `LogicalPlan` and return.
```

Comment:
Every time we select ??? from a view, ??? converts a view to select `LogicalPlan` and returns ???
I use ??? to indicate something missing in this sentence.